### ISSUE TITLE
LLVM assertion showing Dict of Union{}

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#15453]
```
julia> Dict{Union{},Union{}}()
julia: /home/jeff/src/julia-master/julia/deps/srccache/llvm-3.7.1/lib/CodeGen/LiveVariables.cpp:114: void llvm::LiveVariables::MarkVirtRegAliveInBlock(llvm::LiveVariables::VarInfo&, llvm::MachineBasicBlock*, llvm::MachineBasicBlock*, std::vector<llvm::MachineBasicBlock*>&): Assertion `MBB != &MF->front() && "Can't find reaching def for virtreg"' failed.
Aborted (core dumped)
```

Works in 0.4.

**Comment by vtjnash** [comment]
seems to work for me (llvm 3.3 and 3.7.1 on OS X). is this perhaps a llvm regression or linux-specific?

**Comment by Keno** [comment]
I can reproduce this on both Linux/OSX with
```
override LLVM_ASSERTIONS=1
override USE_LLVM_SHLIB=1
```

**Comment by yuyichao** [comment]
With julia-debug

```
julia> Dict{Union{},Union{}}()
Instruction does not dominate all uses!
  %840 = call %jl_value_t* @jl_apply_generic(%jl_value_t** %34, i32 3), !dbg !222
  store %jl_value_t* %840, %jl_value_t** %286, !dbg !234
LLVM ERROR: Broken function found, compilation aborted!
```

**Comment by vtjnash** [comment]
ah, ok. it seems i couldn't reproduce because i was on the https://github.com/JuliaLang/julia/pull/15444 branch (in particular, that branch never uses SSA values directly for variables and lets mem2reg make them SSA only if it considers that possible )

**Comment by JeffBezanson** [comment]
fixed by #15444.

**Comment by tkelman** [comment]
add as a test
